src/ferny/interaction_agent.py

```diff
@@ -30,6 +30,22 @@ class SshError(Exception):
     pass
 
 
+try:
+    recv_fds = socket.recv_fds
+except AttributeError:
+    # Python < 3.9
+    import array
+    import _socket
+
+    def recv_fds(sock, bufsize, maxfds, flags=0):
```

Comment:
If you want to go this approach, we also need `send_fds()` since we use that here.

I wonder if maybe we should just open-code everything, though, with a comment nearby for what we could do in the future... and avoid the hacks with the fallback cases.  We only use this once, so the overhead of doing the shim is worse than just writing it in full in the one place we do it...